test/e2e/testing-manifests/storage-csi/external-provisioner/rbac.yaml

```diff
@@ -21,12 +21,14 @@ apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: external-provisioner-runner
 rules:
-  - apiGroups: [""]
-    resources: ["secrets"]
-    verbs: ["get", "list"]
+  # The following rule should be uncommented for plugins that require secrets
+  # for provisioning.
+  # - apiGroups: [""]
+  #   resources: ["secrets"]
+  #   verbs: ["get", "list"]
   - apiGroups: [""]
     resources: ["persistentvolumes"]
-    verbs: ["get", "list", "watch", "create", "delete"]
+    verbs: ["get", "list", "watch", "update", "create", "delete"]
```

Comment:
Hm I'm curious why the PD CSI tests are not failing due to this?  I updated PD provisioner version yesterday

Reply:
the PD driver has attacher+provisioner in the same pod, so the pod's serviceaccount has update PV permissions already from the attacher rbac https://github.com/kubernetes/kubernetes/blob/master/test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml#L16

Reply:
Ah thanks. We've been contemplating if requiring new rbac permissions should be considered backwards incompatible or not.  I'm a bit on the fence.  We could define it such that if you update the image, you must update the rbacs first. cc @jsafrane @pohly

Reply:
I think it should be considered backwards incompatible, but even if it were defined that way, how do you check or enforce it? I forgot to do it here, tooling to foolproof this would be quite complicated. (& when I changed leader election in the lib I got dozens of reports about permission denied from people like me who updated image w/o also updating RBAC (& it will happen again when the default becomes Leases)).

Reply:
My position is that a new image is compatible with a previous deployment if it can be dropped into that deployment *without* any changes of that deployment (no command line changes, no RBAC rule changes). If it is not compatible, the image has to be released with a major version bump, to warn users that there is some incompatible change.

The CI is going to test this, by dropping new images into unchanged old deployments and running the tests that this deployment is expected to pass.

If the new image can replace the old one with deployment changes and the same features still work (including alpha!) on all Kubernetes versions that we used to support with that deployment, then that's okay. Just because there was a "1.1.0" IMHO is not a guarantee that we'll support that branch. Our guarantee is only that the currently supported Kubernetes versions always have a supported image and that features will not stop working (again, including alpha).

Not removing support for alpha features on older Kubernetes versions is perhaps debatable, but I think it is consistent with what Kubernetes does.